sentry-android-fragment/src/main/java/io/sentry/android/fragment/SentryFragmentLifecycleCallbacks.kt

```diff
@@ -0,0 +1,94 @@
+package io.sentry.android.fragment
+
+import android.content.Context
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentManager.FragmentLifecycleCallbacks
+import io.sentry.Breadcrumb
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SentryLevel.INFO
+
+@Suppress("TooManyFunctions")
+class SentryFragmentLifecycleCallbacks(
+    private val hub: IHub = HubAdapter.getInstance()
```

Comment:
let's also add a ctor overload for better Java compatibility, so the `hub` does not need to be passed at all, similar to https://github.com/getsentry/sentry-java/pull/1491/files

Reply:
Sure, but I think we already have that: there are two constructors for `SentryFragmentLifecycleCallbacks` in Java, and one of them is parameterless. You can check this on [generated API file](https://github.com/getsentry/sentry-java/pull/1522/files#diff-015ad5258f9a132c88f9e5da60e6137c5f55f6b9d031d6776d6742e15b8363e4R23) or here:

<img width="388" alt="image" src="https://user-images.githubusercontent.com/5845095/121654637-39e6c280-ca9e-11eb-9915-2bed4a204d62.png">

Looking how it'd look like in Java: invoking a parameterless constructor from 134 will invoke one from 124 and it'll use the default value: `HubAdapter.getInstance();`

<img width="886" alt="image" src="https://user-images.githubusercontent.com/5845095/121654832-6dc1e800-ca9e-11eb-83c8-bb4d0fd023eb.png">

--- 

I was surprised it works like this but further tests proved that if all arguments in the constructor have a default value, Kotlin will create a parameterless constructor too. I've found it also in docs [here](https://kotlinlang.org/docs/classes.html#secondary-constructors), starting with `On the JVM, if all of`

Is it okay to leave it as it is?

Reply:
indeed, good point, TIL, I just tested and it works, thanks.

Side note (not related to this PR): not all the cases are covered though, just opened this issue: https://github.com/getsentry/sentry-java/issues/1526